internal/services/v1/schema.go

```diff
@@ -143,13 +149,15 @@ func (ss *schemaServer) WriteSchema(ctx context.Context, in *v1.WriteSchemaReque
 			return checkRelErr
 		}
 
-		// Write the new namespaces.
-		if err := rwt.WriteNamespaces(nsdefs...); err != nil {
-			return err
+		// Write the new/changed namespaces.
+		if len(nsDefsWithChanges) > 0 {
+			if err := rwt.WriteNamespaces(nsDefsWithChanges...); err != nil {
+				return err
+			}
 		}
 
 		usagemetrics.SetInContext(ctx, &dispatchv1.ResponseMeta{
-			DispatchCount: uint32(len(nsdefs)),
+			DispatchCount: uint32(len(nsDefsWithChanges)),
 		})
 
 		// Delete the removed namespaces.
```

Comment:
shouldn't the various statements next to this be part of the if statement above

Reply:
No; we still need to set the dispatch count (which could be 0)